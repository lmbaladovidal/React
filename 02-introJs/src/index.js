import heroes from './data/heroes'
console.log(heroes);

const getHeroesById = (id)=>{
    return heroes.find((heroe) => heroe.id === id);
}

const getHeroesByOwner = (owner)=>{
    return heroes.filter((heroe) => heroe.owner === owner);
}

console.log(getHeroesById(2));
console.log(getHeroesByOwner('DC'));
let arr = [0,1,2,3];
arr[10] = 10;
console.log(arr);
console.log(arr[9]);