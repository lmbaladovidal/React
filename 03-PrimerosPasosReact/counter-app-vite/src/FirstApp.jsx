import PropTypes from 'prop-types'
const newMessage = 'Leandro';

function getObj(){
    return {Nombre:"Leandro",
            Apellido:"Balado Vidal"}
}

export const FirstApp = ({title, subtitle,name}) => {
  console.log(title)
  return (
    <>
        <h1 data-testid="test-title">{title}</h1>
        {/* <code>{JSON.stringify(getObj())}</code> */}
        <p>{subtitle}</p>
        <p>{subtitle}</p>
    </>
  )
}

FirstApp.propTypes = {
    title : PropTypes.string.isRequired,
    subtitle : PropTypes.string,
    counter : PropTypes.number
};

FirstApp.defaultProps = {
    name :"Leandro Martin",
    //title : "No Hay titulo",
    subtitle : "No Hay subtitulo"
}
