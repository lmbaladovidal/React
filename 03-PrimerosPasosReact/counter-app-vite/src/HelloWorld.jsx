export const HelloWorld = () => {
  return (
    <h2>HelloWorld</h2>
  )
}
