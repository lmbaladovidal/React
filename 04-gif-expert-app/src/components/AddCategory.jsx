import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('')
    const onInputChange=(event)=>{
        console.log(event.target.value);
        setInputValue(event.target.value)
    }
    const onSubmit = (event)=>{
        event.preventDefault();
        const newInputValue = inputValue.trim() 
        if (newInputValue==0) return;
        /* onAddCategory( categories=>[ inputValue,...categories ]); */
        onNewCategory(newInputValue);
        setInputValue('');

    }
  return (
    <form onSubmit = { onSubmit}>
        <input type="text" value={inputValue } onChange={ onInputChange }/>
    </form>
  )
}
