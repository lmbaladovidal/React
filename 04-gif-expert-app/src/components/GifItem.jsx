
import PropTypes from 'prop-types';

export const GifItem = ({title,url}) => {
  return (
    <>
        <div className="card">
            <img src={url} alt={title?title:"N/A"}></img>
            <p>{title?title:"	༼つಠ益ಠ༽つ ─=≡ΣO))"}</p>
        </div>
    </>
  )
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}

