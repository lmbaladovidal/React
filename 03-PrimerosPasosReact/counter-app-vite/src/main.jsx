import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
import { FirstApp } from './FirstApp';
import {HelloWorld} from './HelloWorld';
import {CounterApp} from './CounterApp';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <HelloWorld />
        <CounterApp value={22}/>
        {/* <FirstApp title={"Hola, soy Goku!"} subtitle='Soy un sexy subtitulo' counter={1}/> */}
    </React.StrictMode>
);