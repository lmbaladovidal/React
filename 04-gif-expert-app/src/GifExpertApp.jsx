import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([ 'Kimetsu no yaiba','My Hero Academy' ]);
  const onAddCategory = (newCategory)=>{
    setCategories([newCategory,...categories]);
  }
  return (
    <>
        {/* titulo */}
        <h1>GifExpertApp</h1>
        {/* input */}
        <AddCategory 
        /* onAddCategory={setCategories} */
        onNewCategory={onAddCategory}
        />
        {/* listado de gif */}
        <ol>          
          {categories.map((category,i)=>{
            return <li key={i}>{category}</li>
          })}
        </ol>
        {/* gif items */}
    </>
  )
}
