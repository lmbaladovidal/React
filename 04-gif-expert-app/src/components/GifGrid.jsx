import { useState, useEffect } from "react";
import { GifItem } from "./GifItem";
import { useFetchGif } from "../hooks/useFetchGif";
//import { getGif } from "../helpers/GetGifs";
export const GifGrid = ({ category }) => {

    const { images,isLoading } = useFetchGif(category) 
    console.log( isLoading )

  return (
    <>
        <h3>{category}</h3>
        {
            isLoading && ( <h2>Cargando...</h2> )
        }
        <div className="card-grid">
            {images.map((image)=>(
                    <GifItem 
                        key={ image.id } 
                        {...image}
                    />
                ))}
        </div>
        
    </>
  )
}
