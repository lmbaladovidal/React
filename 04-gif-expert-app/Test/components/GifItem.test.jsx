import { GifGrid, GifItem } from "../../src/components"
import {render, screen} from '@testing-library/react';

describe('pruebas en <GifItem/>',()=>{
    const title = "Kimetsu";
    const url = "http://kimetsu.com/";
    test('should match with snapshoot', () => { 
        const { container } = render(<GifItem url={url} title={title} />)
        expect(container).toMatchSnapshot();
     })

    test('should the picture with the rigth URL and ATL', () => { 
        render(<GifItem url={url} title={title} />);
        //screen.debug()
        const {src,alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
     })

     test('should display the title in the component', () => { 
        render(<GifItem url={url} title={title} />);
        expect(screen.getByText(title)).toBeTruthy()
      })
})