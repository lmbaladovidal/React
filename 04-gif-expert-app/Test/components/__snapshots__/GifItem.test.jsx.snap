// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem/> should match with snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Kimetsu"
      src="http://kimetsu.com/"
    />
    <p>
      Kimetsu
    </p>
  </div>
</div>
`;
